gnomad_qc/v4/subset.py

```diff
@@ -0,0 +1,222 @@
+            "Number": "1",
+            "Type": "Integer",
+        },
+        "PGT": {
+            "Description": "Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another",
+            "Number": "1",
+            "Type": "String",
+        },
+        "PID": {
+            "Description": "Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group",
+            "Number": "1",
+            "Type": "String",
+        },
+        "PL": {
+            "Description": "Normalized, phred-scaled likelihoods for genotypes as defined in the VCF specification",
+            "Number": "G",
+            "Type": "Integer",
+        },
+        "SB": {
+            "Description": "Per-sample component statistics which comprise the Fisher's exact test to detect strand bias. Values are: depth of reference allele on forward strand, depth of reference allele on reverse strand, depth of alternate allele on forward strand, depth of alternate allele on reverse strand.",
+            "Number": "4",
+            "Type": "Integer",
+        },
+        "RGQ": {
+            "Description": "Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)"
+        },
+    },
+}
+
+
+def main(args):
+    hl.init(log="/subset.log", default_reference="GRCh38")
+    header_dict = HEADER_DICT
+    test = args.test
+
+    vds = get_gnomad_v4_vds(n_partitions=args.n_partitions)
+
+    if test:
+        vds = hl.vds.variant_dataset.VariantDataset(
+            vds.reference_data._filter_partitions(range(2)),
+            vds.variant_data._filter_partitions(range(2)),
+        )
+
+    meta_ht = project_meta.ht()
+    subset_ht = hl.import_table(args.subset or args.subset_workspaces)
+
+    if args.subset_workspaces:
+        terra_workspaces = hl.literal(subset_ht.terra_workspace.lower().collect())
+        subset_ht = meta_ht.filter(
+            terra_workspaces.contains(meta_ht.project_meta.terra_workspace)
+        ).select()
+        logger.info("Keeping %d samples using the list of terra workspaces.", subset_ht.count())
+
+    if args.include_ukb_200k:
+        ukb_subset_ht = meta_ht.filter(
+            (meta_ht.project_meta.terra_workspace == "ukbb_wholeexomedataset")
+            & hl.literal(UKB_BATCHES_FOR_INCLUSION).contains(
+                meta_ht.project_meta.ukb_meta.ukb_batch
+            )
+        ).select()
+        subset_ht = subset_ht.union(ukb_subset_ht)
+        logger.info("Keeping %d samples after inclusion of the UKB 200K subset.", subset_ht.count())
+
+    vds = hl.vds.filter_samples(vds, subset_ht, remove_dead_alleles=True)
+
+    if args.include_ukb_200k:
+        # TODO: add option to provide an application linking file as an argument. Default is ATGU ID
+        vds = hl.vds.variant_dataset.VariantDataset(
```

Comment:
I think this will be whatever we add to an application linking file as stated in the TODO, just trying to get the script in  so that we can return a VDS subset quickly